frontend/amundsen_application/static/js/pages/SearchPage/SearchFilter/InputFilter/index.tsx

```diff
@@ -63,8 +63,11 @@ export class InputFilter extends React.Component<
 
     this.setState({ value: newValue });
 
-    filterState[resourceType][categoryId] = newValue;
-    updateFilterState(filterState);
+    const newFilters = {
+      ...filterState,
+      [resourceType]: { ...filterState[resourceType], [categoryId]: newValue },
+    };
```

Comment:
Yep, mutation is better to be avoided. I hope it didn't take you too long to find this out!